### PR TITLE
Tweak ADDing/COPYing directories to copy contents

### PR DESCRIPTION
**Pull request by nalind**
When copying or adding a source directory, copy the directory's contents to the destination directory, to better match Dockerfile COPY behavior.

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

File: tests/copy.bats [hunk 1]
```diff
@@ -45,3 +45,24 @@ load helpers
...
+  test -s $root/container-subdir/randomfile
+  cmp ${TESTDIR}/subdir/randomfile $root/container-subdir/randomfile
+  test -s $root/container-subdir/other-randomfile
+  cmp ${TESTDIR}/subdir/other-randomfile $root/container-subdir/other-randomfile
+  buildah copy $cid ${TESTDIR}/subdir /other-subdir
+  test -s $root/other-subdir/randomfile
+  cmp ${TESTDIR}/subdir/randomfile $root/other-subdir/randomfile
+  test -s $root/other-subdir/other-randomfile
+  cmp ${TESTDIR}/subdir/other-randomfile $root/other-subdir/other-randomfile
+  buildah delete $cid
```
Comment (nalind) [inline tests/copy.bats]: Whoops, this should be changed to "buildah rm".

Comment (rhatdan) [inline tests/copy.bats]: buildah delete would work though.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+ 8334198

**Comment by nalind** [comment]
[[Buildah] Add "dockah build-dockerfile"](https://trello.com/c/AyBXqoRY/443-buildah-add-dockah-build-dockerfile)

**Pull request closed without merging**